### DISCUSSION TITLE
Exclude RIC on client

### DISCUSSION THREAD

**Discussion by sm7tix** [BOSWatch/BW3-Core#124] [Ideas]
I have some RICs on my clients that send encrypted or crappy Pocsag messages. 
It have been nice to be able to put it in the client.yaml file on a raw xalled exlric or something. 
On my clients i have done this manually by adding some code to 
/opt/boswatch3/boswatch/decoder/pocsagDecoder.py

if re.search("[0-9]{7}", ric) and re.search("[1-4]", subric):
            if "Alpha:" in data:
                if ric == '1600000':
                    logging.info("Ric 1600000")
                    return
                if ric == '0682408':
                    logging.info("Ric 0682408")
                    return

I do this because i have the server on other location and i do not like the client to send crapp to server. This code skippes the ric i have added. 

To be able to have a loop here picking RIC from client.yaml should be nice to have,. 
Best regards Stefan / Sm7Tix

**Comment by KoenigMjr** [comment]
i know its a bit late for you, but maybe others can find help...

should be possible with regex-module..

As Schrolli also replied at #86:

> Take a look at the modules - especially the regex filters maybe ;-) https://docs.boswatch.de/modul/regex_filter.html
> 
> You can use the modules inside the router configuration - see here for example: https://docs.boswatch.de/information/router.html`

Is it possible (i think! not tested, maybe you can give it a shot?) that you can use negative lookahead in regex. For your two Rics it should be something like:

`^(?!1600000$|0682408$).*`

Explanation:
^ = Start of line
(?!...) = negative lookahead (exclude)
1600000$|0682408$ = means: exactly like this ($ means end of line; | separator)
.* = everything else

if you want to extend the exclucions, just add `|1234567$` to the rics, so it gets `^(?!1600000$|0682408$|1234567$).*`

kind regards
